apps/sms/js/threads.js

```diff
@@ -86,12 +88,19 @@
 
   var Threads = exports.Threads = {
     registerMessage: function(message) {
-      var thread = Thread.create(message);
       var threadId = message.threadId;
+
       if (!this.has(threadId)) {
-        this.set(threadId, thread);
+        this.set(threadId, Thread.create(message));
+      }
+      this.get(threadId).messages.set(message.id, message);
+    },
+    unregisterMessage: function(message) {
```

Comment:
I thought you could use the same line you currently use before calling unregisterMessage but I missed that you use Threads.active which we could not do here.

Can we have another Map for messages ? This wouldn't take much more memory because we'd have the same "message" object.